cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/BudgetsPage.java

```diff
@@ -0,0 +1,324 @@
+ * - confirmDeleteButton: The button to confirm the deletion of a budget.
+ * 
+ * Edit and Delete Savings Bucket Modal Elements:
+ * - editSavingsBucketButton: The button to edit a savings bucket.
+ * - deleteSavingsBucketButton: The button to delete a savings bucket.
+ * - editedAmountRequiredInput: The input field for the edited amount required.
+ * - saveSavingsBucketButton: The button to save the edited savings bucket.
+ * - confirmDeleteSavingsBucketButton: The button to confirm the deletion of a savings bucket.
+ * 
+ * Methods:
+ * - BudgetsPage(WebDriver driver): Constructor to initialize the BudgetsPage with the given WebDriver.
+ * - getChildComponents(): Returns a list of child components.
+ * - getChildComponent(String name): Returns a child component by name.
+ * - getWebElements(): Returns a list of web elements.
+ * - getWebElement(String name): Returns a web element by name.
+ * - getButtons(): Returns a list of buttons.
+ * - clickButton(String name): Clicks a button by name.
+ * - setGroceryActionButtons(String budgetCategory, String budgetedAmount): Sets the action buttons for a grocery budget.
+ * - setEditBudgetModalElements(): Sets the elements for the edit budget modal.
+ * - setDeleteButtonForGroceriesBudget(): Sets the delete button for the groceries budget.
+ * - setSavingsBucketActionButtons(String bucketName, String amount): Sets the action buttons for a savings bucket.
+ * - setEditSavingsBucketModalElements(): Sets the elements for the edit savings bucket modal.
+ * - setDeleteButtonForHouseRepairsSavingsBucket(): Sets the delete button for the house repairs savings bucket.
+ */
+
+public class BudgetsPage extends Page {
+    // Components
+    @FindBy(id = "summary-component")
+    private WebElement summaryComponentDiv;
+
+    @FindBy(id = "Add-New-Budget")
+    private WebElement addNewBudgetButton;
+
+    @FindBy(id = "totalAmount")
+    private WebElement budgetedInput;
+
+    @FindBy(xpath = "//*[@id=\"root\"]/div[1]/main/table[1]")
+    private WebElement budgetsTable;
+
+    @FindBy(xpath = "//*[@id=\"root\"]/div[1]/main/table[2]")
+    private WebElement savingsBucketTable;
+
+    @FindBy(id = "Budget-Name-Input")
+    private WebElement budgetNameInput;
+
+    @FindBy(id = "Amount-Required-Input")
+    private WebElement amountRequiredInput;
+
+    @FindBy(id = "Submit-New-Savings-Bucket")
+    private WebElement submitNewSavingsBucketButton;
+
+    @FindBy(id = "Edit-Spending-Budget")
+    private WebElement editSpendingBudgetButton;
+
+    @FindBy(id = "totalBudgetAmount")
+    private WebElement totalBudgetAmountInput;
+
+    @FindBy(id = "Submit-Spending-Budget")
+    private WebElement submitSpendingBudgetButton;
+
+    @FindBy(id = "Spending-Budget-Div")
+    private WebElement spendingBudgetDiv;
+
+    @FindBy(css = "div.nav-container.bg-\\[\\#005ea2\\].px-6.h-screen.min-w-64.max-w-min-w-64.flex.flex-col.pt-6")
+    private Navbar navbar;
```

Comment:
Incorrect usage: FindBy is meant for WebElements, while navbar is a Component.
Suggestion: Remove FindBy and initialize the Navbar in the BudgetsPage constructor. Setting it as either the LandingNavbar or DashboardNavbar should cover all intended purposes (likely want to use DashboardNavbar)